api/src/test/java/marquez/db/SearchDaoTest.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import marquez.api.models.SearchFilter;
+import marquez.api.models.SearchResult;
+import marquez.api.models.SearchSort;
+import marquez.jdbi.MarquezJdbiExternalPostgresExtension;
+import org.glassfish.jersey.internal.guava.Lists;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/** The test suite for {@link SearchDao}. */
+@Tag("IntegrationTests")
+@ExtendWith(MarquezJdbiExternalPostgresExtension.class)
+public class SearchDaoTest {
+  static final int LIMIT = 25;
+  static final int NUM_OF_JOBS = 2;
+  /**
+   * Using {@link DbTestUtils#newJobs(Jdbi, int)} generates 4 input datasets and 3 output datasets
+   * for each job inserted into the {@code jobs} table.
+   */
+  static final int NUM_OF_DATASETS = 12;
+
+  static SearchDao searchDao;
+
+  @BeforeAll
+  public static void setUpOnce(final Jdbi jdbi) {
+    searchDao = jdbi.onDemand(SearchDao.class);
+
+    DbTestUtils.newDataset(jdbi, "name_ordering_0");
+    DbTestUtils.newDataset(jdbi, "name_ordering_1");
+    DbTestUtils.newDataset(jdbi, "name_ordering_2");
+
+    DbTestUtils.newDataset(jdbi, "time_ordering_0");
+    DbTestUtils.newDataset(jdbi, "time_ordering_1");
+    DbTestUtils.newDataset(jdbi, "time_ordering_2");
+
+    DbTestUtils.newJobs(jdbi, NUM_OF_JOBS);
+  }
+
+  @Test
+  public void testSearch() {
+    final String query = "test";
+    final List<SearchResult> results = searchDao.search(query, null, SearchSort.NAME, LIMIT);
+
+    // Ensure search results contain N datasets and M jobs.
+    assertThat(results).hasSize(NUM_OF_DATASETS + NUM_OF_JOBS);
+
+    // Group search results by result type.
+    final List<List<SearchResult>> resultsGroupedByType = groupResultsByType(results);
+
+    // Ensure search results contain exactly N datasets.
+    final List<SearchResult> resultsWithOnlyDatasets = resultsGroupedByType.get(0);
```

Comment:
I don't think you can always rely on this ordering- datasets might be first or it might be second. I'd return a map keyed by `result.getType()` then get the list for the `SearchResult.ResultType.DATASET` key